py4web/utils/form.py

```diff
@@ -819,96 +839,100 @@ def __init__(
-                                            value.filename,
-                                            field.uploadfolder,
-                                        )
-                                    elif field.uploadfield and field.db:
-                                        validated_vars[field.name] = field.store(
-                                            value.file,
-                                            value.filename,
-                                            field.uploadfolder,
-                                        )
-                                    else:
-                                        validated_vars[field.name] = value
-                                elif self.record:
-                                    if not delete:
-                                        validated_vars[field.name] = self.record.get(
-                                            field.name
-                                        )
-                                    else:
-                                        validated_vars[field.name] = value = None
-                            elif field.type == "boolean":
-                                validated_vars[field.name] = value is not None
-                            else:
-                                validated_vars[field.name] = value
-                            if error:
-                                self.errors[field.name] = error
-                    if self.errors:
-                        for field_name in uploaded_fields:
-                            validated_vars[field_name] = (
-                                self.record and self.record.get(field_name) or None
-                            )
-                    self.vars.update(validated_vars)
-                    if self.record and dbio:
-                        self.vars["id"] = self.record.id
-                    if validation:
-                        validation(self)
-                    if not self.errors:
-                        self.accepted = True
-                        if dbio:
-                            self.update_or_insert(validated_vars)
-                elif dbio:
-                    self.accepted = True
-                    self.deleted = True
-                    self.record.delete_record()
+        try:
+            self.post_vars = request.GET if self.method == "GET" else request.POST
+        except KeyError:
+            self.post_vars = {}
+
+        try:
+            form_vars = copy.deepcopy(request.forms)
+        except KeyError:
+            form_vars = {}
+        for k in form_vars:
+            self.vars[k] = form_vars[k]
+
+        # We only a process a form if it is POST and the formkey matches (correct formname and crsf)
+        # Notice: we never expose the crsf uuid, we only use to sign the form uuid
+        if self.post_vars:
+            self.submitted = True
+
+        if not readonly and auto_process:
+            self.process()
         if self.csrf_protection:
```

Comment:
Should this now always run after process() even if auto_process is False?

Reply:
oh, you're right